File: 2010_2011_mf_sujet_4_convergences_de_processus/plotergo.py
import numpy as np
import matplotlib.pyplot as plt

from Fattente import Fattente
from plim import plim


def plotergo(p, q, nmax):
    plt.clf()
    X = np.zeros(nmax, dtype=int)
    for k in range(1, nmax):
        X[k] = Fattente(X[k - 1], p, q)
    plimite = plim(p, q)
    fig, axes = plt.subplots(3, 5, figsize=(15, 9))
    for m in range(15):
        ax = axes[m // 5, m % 5]
        erg = np.cumsum(X == m) / np.arange(1, nmax + 1)
        ax.plot(erg)
        ax.axhline(plimite[m], color='r')
        ax.set_title(f'm={m}')
    plt.tight_layout()

File: 2010_2011_mf_sujet_4_convergences_de_processus/plotergo.m
function  plotergo(p,q,nmax )
%UNTITLED8 Summary of this function goes here
%   Detailed explanation goes here
clf;
X=zeros(1,nmax);
X(1)=0;
for k=2:nmax
    X(k)=Fattente(X(k-1),p,q);
end;
plimite=plim(p,q);
Y=zeros(15,nmax);
for m=0:14
    Y(m+1,:)=cumsum(X==m)./[1:nmax];
end
for m=1:15
    subplot(3,5,m); hold on;
    plot(Y(m,:));
    plot(plimite(m)*ones(1,nmax),'r');hold off;
end



end

